app/serializers/oauth_userinfo_serializer.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class OauthUserinfoSerializer < ActiveModel::Serializer
+  include RoutingHelper
+
+  attributes :iss, :sub, :name, :preferred_username, :profile, :picture
+
+  def iss
+    root_url
+  end
+
+  def sub
+    ActivityPub::TagManager.instance.uri_for(object)
+  end
+
+  def name
+    object.unavailable? ? '' : object.display_name
+  end
+
+  def preferred_username
+    object.username
+  end
+
+  def profile
+    ActivityPub::TagManager.instance.url_for(object)
+  end
+
+  def picture
+    full_asset_url(object.unavailable? ? object.avatar.default_url : object.avatar_original_url)
+  end
```

Comment:
This is ok but it seems unnecessary. In practice the controller requires a functional account, so `object.unavailable?` will always be true.